src/sentry/static/sentry/app/components/dropdownLink.jsx

```diff
@@ -1,98 +1,169 @@
-import jQuery from 'jquery';
 import PropTypes from 'prop-types';
 import React from 'react';
 import classNames from 'classnames';
 
-require('bootstrap/js/dropdown');
-
-const DropdownLink = React.createClass({
-  propTypes: {
+class DropdownLink extends React.Component {
+  static propTypes = {
     title: PropTypes.node,
     /** display dropdown caret */
     caret: PropTypes.bool,
     disabled: PropTypes.bool,
     onOpen: PropTypes.func,
     onClose: PropTypes.func,
+    /**
+     * If this is set, then this will become a "controlled" component.
+     * It will no longer set local state and dropdown visiblity will
+     * only follow `isOpen`.
+     */
+    isOpen: PropTypes.bool,
     /** anchors menu to the right */
     anchorRight: PropTypes.bool,
+    /** Keeps dropdown menu open when menu is clicked */
+    keepMenuOpen: PropTypes.bool,
     topLevelClasses: PropTypes.string,
     menuClasses: PropTypes.string
-  },
+  };
 
-  getDefaultProps() {
-    return {
-      disabled: false,
-      anchorRight: false,
-      caret: true
-    };
-  },
-  getInitialState() {
-    return {
+  static defaultProps = {
+    disabled: false,
+    anchorRight: false,
+    caret: true
+  };
+
+  constructor(...args) {
+    super(...args);
+    this.state = {
       isOpen: false
     };
-  },
-
-  componentDidMount() {
-    jQuery(this.refs.dropdownToggle).dropdown();
-    jQuery(this.refs.dropdownToggle.parentNode)
-      .on('shown.bs.dropdown', e => {
-        this.setState({
-          isOpen: true
-        });
-        this.props.onOpen && this.props.onOpen(e);
-      })
-      .on('hidden.bs.dropdown', e => {
-        setTimeout(() => {
-          if (!this.isMounted()) {
-            return;
-          }
-          this.setState({
-            isOpen: false
-          });
-          this.props.onClose && this.props.onClose(e);
-        });
-      });
-  },
+  }
 
   componentWillUnmount() {
-    jQuery(this.refs.dropdownToggle.parentNode).off();
-  },
+    $(document).off('click', this.checkClickOutside);
+  }
+
+  // Gets open state from props or local state when appropriate
+  isOpen = () => {
+    let {isOpen} = this.props;
+    let isControlled = typeof isOpen !== 'undefined';
```

Comment:
we have a defined function in utils.jsx

Reply:
it checks for undefined or null

Reply:
I want a strict undefined check, i.e `<DropdownLink />` and `<DropdownLink isOpen={null} />` should be treated differently,